src/app/services/storage/storage.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from '@angular/core';
+import { Storage } from '@ionic/storage';
+import { StorageKey } from './storage-key';
+import { LimitedCacheOptions } from './storage-options';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StorageService {
```

Comment:
Refactor and make methods more readable. The limitedArrayPush method is atrocious.